### ISSUE TITLE
Lane assist arrows shape don't match backend instructions

### ISSUE THREAD

**Issue by Penegal** [Project-OSRM/osrm-frontend-v2#281]
Hello, there.

![image](https://user-images.githubusercontent.com/13777824/42016506-1124370a-7aac-11e8-8caf-7e8c560a26f2.png)
![image](https://user-images.githubusercontent.com/13777824/42016526-25034b12-7aac-11e8-9520-78095bff6ba9.png)

As you can see here (http://map.project-osrm.org/?z=18&center=48.192663%2C6.438981&loc=48.192459%2C6.436872&loc=48.192670%2C6.439394&hl=fr&alt=0), the arrows displayed by the backend do not have the shape the backend gave: the highway is tagged `turn:lanes=left;through|through;right`, but the lane assist is as if the value was `through|through`.

Regards.

**Comment by yuryleb** [comment]
This is due to frontend restriction - it can't display "complex" arrows into more than one direction.
If there are several "arrows" for one lane, frontend just tries to choose the more [appropriate one](https://github.com/Project-OSRM/osrm-frontend/blob/gh-pages/src/itinerary_builder.js#L41-L58) from them.

**Comment by Penegal** [comment]
What's the problem? Missing icons for arrows?

**Comment by danpat** [comment]
@Penegal basically, yes.  The API response reports the arrow components, like `through;right`, but we only have icons for `through` and `right` individually.  We would need to create a combined `through_right` icon, or combine both images at display time.

(There's a further step that would be nice which would be to highlight the maneuver that is valid for a given lane, even if it's a composite symbol, but our `lanes` only indicates which lanes are valid, not which symbols on each lane are valid).

**Comment by Penegal** [comment]
@danpat: composing the combined icon from individual arrows would allow to easily highlight, on a composed one, only the arrow to be followed. If you have arrow icons with alpha, designed to be stacked on display, and with a highlighted and a gray icon for each arrow direction, then you only have to stack them on display to get any composed arrow you want.

If you also want, in the composed arrow, to highlight only the direction to be followed, then you only have to put the corresponding highlighted arrow symbol above the grayed-out symbols of the other arrows composing the combined one. That is similar to what is requested in #192 and Project-OSRM/osrm-backend#2935. AFAIK, the tricky part is to design the icons to be exactly stacked on display, without shifting the line of the arrow on display when stacked, but I don't know enough web or OSRM development to know the difficulty of implementing this feature, and I assume this is not as easy as I see.

**Comment by yuryleb** [comment]
I'm not web designer too but it seems I find the way to draw lane icons over each other:
![ovelapped icons](https://user-images.githubusercontent.com/4529411/42125757-16222278-7c6c-11e8-820f-2ddfb66a7737.png)
The second lane with two allowed directions looks in HTML as:
```HTML
<td>Take the ramp towards <b>I 66 West: E Street</b>
 <br/>
 <span class="leaflet-routing-icon lanes invalid leaflet-routing-icon-continue"></span>
 <span class="leaflet-routing-icon lanes invalid leaflet-routing-icon-continue"></span>
 <span class="leaflet-routing-icon lanes leaflet-routing-icon-turn-right" style="left: -14px; position: relative;"></span>
</td>
```
Now it's just necessary to redraw all icons to be of the same width and centered on the same axis :wink:

**Comment by Penegal** [comment]
I think I can take care of this, but, as I'm not a web developper, could you please tell me where these icons are, so I can give you the corrected versions?

**Comment by yuryleb** [comment]
Actually I already created new icons file with lane indications only and based on `images/osrm.directions.icons.color.ai/.svg` , just still checking how it works :wink:

**Comment by Penegal** [comment]
Ok, then.

**Comment by Penegal** [comment]
This has been solved by #284.
